keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+- Allowing time for feedback
+- Completion of all accepted "future work" items
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+#### Upgrade considerations
+
+As such, the cluster-scoped policy resources are new and shall not exist prior
+to upgrading to a new version. Thus, there is no direct impact on upgrades.
+
+#### Downgrade considerations
+
+Downgrading to a version which no longer supports cluster-scoped policy APIs
+must ensure that appropriate security rules are created to mimick the cluster-scoped
+policy rules by other means, such that no unintended traffic is allowed, and all
+intended traffic is allowed.
+
+### Version Skew Strategy
+
+n/a
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+1.24:
+- disable by default
+- allow gate to enable the feature
+- release note
+
+1.26:
+- enable by default
+- allow gate to disable the feature
+- release note
+
+1.28:
+- remove gate
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: AdminNetworkPolicy
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature by itself has no effect on the cluster.
+Creating a AdminNetworkPolicy does have an effect on
+the cluster, however they must be specifically created, which means the
```

Comment:
You also need an implementation that supports this feature. With NetPol today, as I recall we have no way the user can tell whether a given cluster supports the feature. Is that something we can do better with in this feature?

Reply:
Yes, we now have a `Status` Field which will allow the implementation to report weather of not the `AdminNetworkPolicy` Resource is supported